app/controllers/application_controller.rb

```diff
@@ -223,7 +223,7 @@ def execute_file_download(study)
     begin
       # get filesize and make sure the user is under their quota
       requested_file = ApplicationController.firecloud_client.execute_gcloud_method(:get_workspace_file, 0, study.bucket_id, params[:filename])
-      if requested_file.present?
+      if requested_file.present? && !requested_file.size.nil?
```

Comment:
```suggestion
      if requested_file.present? && requested_file.size.present?
```
is a little more readable

Reply:
> Looks good, confirmed via local manual test.
> 
> Cavaet: redirect fails if study name is omitted in URL. E.g. https://localhost:3000/single_cell/data/public/1/the-study-name works, but https://localhost:3000/single_cell/data/public/SCP1/ fails. I see no cases like the latter in the linked Sentry report. So while accounting for such cases would be a nice touch, I think the current update suffices.

Just to explain the problem:
The sentry error was due to the filename missing in the URL. This [line of code](https://github.com/broadinstitute/single_cell_portal_core/blob/c593b6bbc781dc582dcf402e88e92df11ca62c7b/app/controllers/application_controller.rb#L225) passes the file name from the URL. When the file name isn't in the URL, the variable `requested_file` is an empty object. As a result, [file_name](https://github.com/broadinstitute/single_cell_portal_core/blob/c593b6bbc781dc582dcf402e88e92df11ca62c7b/app/controllers/application_controller.rb#L227) is nil. When trying to calculate the [user_quota](https://github.com/broadinstitute/single_cell_portal_core/blob/c593b6bbc781dc582dcf402e88e92df11ca62c7b/app/controllers/application_controller.rb#L228) this causes an error because you can not add an integer to a nil value. This is why we saw the error in Sentry. I'm not sure what behavior would ensue with the URL you tested against. 


> Looks good, confirmed via local manual test.
> 
> Cavaet: redirect fails if study name is omitted in URL. E.g. https://localhost:3000/single_cell/data/public/1/the-study-name works, but https://localhost:3000/single_cell/data/public/SCP1/ fails. I see no cases like the latter in the linked Sentry report. So while accounting for such cases would be a nice touch, I think the current update suffices.